### ISSUE TITLE
Provide support for GitHub Actions Job summaries

### ISSUE THREAD

**Issue by jrhemstad** [Kitware/CDash#1765]
# Feature Request

### How can we make CDash better?

I use GitHub actions workflows for all of my continuous build/test for my C++ library using cmake/ninja/ctest. I would like to be able to leverage CDash's log parsing and reporting more natively within GitHub. 

One of the primary benefits of using GitHub Actions is how tightly integrated it is with GitHub and developers don't have to leave GitHub to see the results of the build/test matrix. One weakness of GitHub Actions is the build/test logs are not very digestible by default. 

CDash is excellent at parsing build/test logs and highlighting the various failures, but requires navigating to an external site and tool in order to view the matrix summary as well as each jobs build/test summary. 

I want to have my cake and eat it too by making CDash's build/test job summary available directly in the GitHub UI. From what I can tell, the easiest way to do this would be via a [custom job summary](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/). This lets you provide arbitrary Markdown which will be rendered in the job summary page. 

In short, I'd want to be able to provide the equivalent of [this CDash build summary](https://open.cdash.org/viewBuildError.php?buildid=9106276) but as a GitHub Job Summary that renders natively in the GitHub UI.
